### ISSUE TITLE
Wells created/modified via ActionX not treated correctly in parallel.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#3757]
Triggered by OPM/opm-grid#564.

It is highly doubtful that modifying wells with ActionX works in parallel for all cases with flow. (Let's hope someone proves me wrong...)

The problem is that the simulator assumes that the information about all possible perforations is available when starting the simulator (`Schedule::getWellatEnd()`) and that the distribution of wells is static as in "no new perforations appear.

At the moment my assumption are rather theoretic, meaning that I have not checked the code fully.

## Loadbalance approach that keeps all perforations of a well on one process

This is the default in flow. Before load balancing we inspect the return value of `Schedule::getWellatEnd()` and use that information to make sure that no well is split. Unfortunately, neither new perforations added to wells or new wells added via ActionX will be taken into account.

+ If an added perforation is not part of the interior on the same process where the well lives, it will at least be skipped on that process. Maybe another well with just that connection might appear on another process without knowledge of this one. Calculations will be wrong.
+ If the new perforation is by chance on the same process, we are lucky and safe.
+ If a new well appears that is part of the interior of several processes, we might not do the correct thing. I am certain about this if running with --matrix-add-well-contributions=true. For the other case we need to be check the code, maybe we are luck (famous last words).
+ If the new well is in the interior of one process, everything probably works.

## Loadbalance approach that supports distributed wells

T.B.C

**Comment by blattms** [comment]
@lisajulia Would you comment here what of this is fixed now and what still might be an issue? Thanks a lot.
